sprint_19

Fennec-meow opened the pull request:
Changes:

1. The "Launch Screen" is configured;
2. The main application screen is configured with a choice of the city and station of departure and destination;
3. Set up a screen with a list of cities (searchable);
4. A screen with a list of train stations (stations) in the specified city is configured (searchable);
5. A screen with a list of carriers is set up;
6. The filter screen is set up (opens by pressing the "Specify time" button);
7. The screens for the situations "Server error" and "No Internet" are configured.

Fennec-meow committed:
Changes:

1. The "Launch Screen" is configured;
2. The main application screen is configured with a choice of the city and station of departure and destination;
3. Set up a screen with a list of cities (searchable);
4. A screen with a list of train stations (stations) in the specified city is configured (searchable);
5. A screen with a list of carriers is set up;
6. The filter screen is set up (opens by pressing the "Specify time" button);
7. The screens for the situations "Server error" and "No Internet" are configured.

Fennec-meow committed:
Fix

IgorChumikov commented on TravelSchedule/ChoosingCityView/ChoosingCityView.swift:
```diff
@@ -0,0 +1,211 @@
+
+// MARK: - ChoosingCityView
+
+struct ChoosingCityView: View {
```
👨‍🏫 Рекомендация по структуре проекта:

Сейчас в ChoosingCityView.swift находится сразу несколько представлений (CityScrollView, ListCities, CityNotFound, BackButton, CitySelectionTitle), что затрудняет читаемость и поддержку кода.

✅ Лучше вынести каждое view в отдельный файл:
Это даст следующие преимущества:

Повышается читаемость и масштабируемость проекта;
Проще тестировать и переиспользовать компоненты;
Уменьшается размер одного файла;
Код легче ревьюить.

🔧 Пример:
ChoosingCityView.swift – только главный экран;
CityScrollViewView.swift – скролл с городами;
ListCitiesView.swift – список кнопок с городами;
CityNotFoundView.swift – заглушка «Город не найден»;
BackButtonView.swift – кнопка назад;
CitySelectionTitleView.swift – заголовок.

📝 Совет по неймингу:
Если файл содержит SwiftUI-представление (View), обязательно указывай в названии View в конце.
Например: CityNotFoundView.swift — так сразу видно, что это View, а не модель или сервис.

IgorChumikov commented on TravelSchedule/FlightSelectionView/FlightSelectionView.swift:
```diff
@@ -0,0 +1,42 @@
+    
+    var body: some View {
+        VStack {
+            
```
Лишний отступ

IgorChumikov commented on TravelSchedule/Model/NavCoordinator.swift:
```diff
@@ -0,0 +1,24 @@
+
+// MARK: - NavCoordinator
+
+final class NavCoordinator: ObservableObject {
```
📦 Замечание по структуре проекта:

Сейчас файл NavCoordinator.swift находится в папке Model, но по сути он не является моделью данных. Это объект, который управляет состоянием навигации и выбором маршрута, то есть скорее относится к слою ViewModel или Coordinator.

🔹 Navigation / Coordinators
Если ты используешь координаторы как отдельный слой архитектуры, можно создать папку Navigation или Coordinators:
```

TravelSchedule
├── Views
├── Coordinators ← альтернатива
│   └── RouteFlowCoordinator.swift

```

IgorChumikov commented on TravelSchedule/SettingsView/SettingsView.swift:
```diff
@@ -0,0 +1,23 @@
+
+// MARK: - SettingsView
+
+struct SettingsView: View {
```
Сейчас SettingsView содержит только вызов ServerErrorScreen() и не выполняет никакой другой логики. Из-за этого непонятно, зачем вообще нужен этот экран и какое он имеет отношение к настройкам.

IgorChumikov commented on TravelSchedule/TestFetch/testFetch.swift:
```diff
@@ -0,0 +1,239 @@
+import OpenAPIURLSession
+import OpenAPIRuntime
+
+struct testFetch: View {
```
🛠️ Замечание по стилю кода:

Классы, структуры и перечисления в Swift должны начинаться с заглавной буквы по стандарту Swift API Design Guidelines.

Fennec-meow committed:
Changes:

1. CitySelectionScreen - fixed the display of the "From and To" departure button and lines;
2. StationSelectionView - when selecting a station, it goes to the main screen;
3. SearchTextField - the cancel button for the entered text appears only if the text is entered in the TextField;
4. testFetch - renamed to "TestFetch".

Fennec-meow committed:
Fix

Fennec-meow committed:
Fix

Fennec-meow committed:
Fix

Fennec-meow committed:
Fix

Fennec-meow committed:
Fix

Fennec-meow committed:
Fix

Fennec-meow committed:
Fix

Fennec-meow committed:
Fix

Fennec-meow committed:
Fix
